src/common/lib/util/utils.ts

```diff
@@ -547,3 +547,26 @@ export function shallowEquals(source: Record<string, unknown>, target: Record<st
     Object.keys(target).every((key) => target[key] === source[key])
   );
 }
+
+export function matchDerivedChannel(name: string) {
+  /**
+   * This regex check is to retain existing channel params if any e.g [?rewind=1]foo to
+   * [filter=xyz?rewind=1]foo. This is to keep channel compatibility around use of
+   * channel params that work with derived channels.
+   *
+   * This eslint unsafe regex warning is triggered because the RegExp uses nested quantifiers,
+   * but it does not create any situation where the regex engine has to
+   * explore a large number of possible matches so it’s safe to ignore
+   */
+  const regex = /^(\[([^?]*)(?:(.*))\])?(.+)$/; // eslint-disable-line
+  const match = name.match(regex);
+  if (!match || !match.length || match.length < 5) {
+    throw new ErrorInfo('regex match failed', 400, 40010);
+  }
+  // Fail if there is already a channel qualifier, eg [meta]foo should fail instead of just overriding with [filter=xyz]foo
+  if (match![2]) {
+    throw new ErrorInfo(`cannot use a derived option with a ${match![2]} channel`, 400, 40010);
+  }
+  // Return match values to be added to derive channel quantifier.
+  return `${match![3] || ''}]${match![4]}`;
```

Comment:
It seems a unexpected to return a string here with a closing bracket `]` in the middle of it. I would suggest we returned two values `match![3] || ''` and `match![4]` in an object with reasonable names as keys (e.g. `qualifierParam` and `channelName`), and then use those from the caller and make the caller responsible for closing their parentheses.